tests/TestCase.php

```diff
@@ -192,4 +204,19 @@ protected function createMockWithMethods($class, $methods = [])
         }
         return $mock;
     }
+
+    /**
+     * @param $classInstance
+     * @param $dependencyName
+     * @param $newDependency
+     * @return void
+     * @throws \ReflectionException
+     */
+    protected function replaceDependency($classInstance, $dependencyName, $newDependency)
```

Comment:
Very usefull I think, let's see if we can refactor our tests with it in the future :+1: